workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/src/CodeGenerator.cs

```diff
@@ -61,11 +62,15 @@ public int Run()
             var schemaFilesRaw = SchemaFiles.GetSchemaFilesRaw(options.JsonDirectory, fileSystem).ToList();
             var schemaProcessor = new UnitySchemaProcessor(schemaFilesRaw);
             var globalEnumSet = ExtractEnums(schemaProcessor.ProcessedSchemaFiles);
-
+            var workerTypes = ExtractWorkerTypes(options.WorkerJsonDirectory);
+            
+            var workerGenerationJob = new WorkerGenerationJob(options.NativeOutputDirectory, fileSystem, workerTypes);
```

Comment:
Pass in the whole options data blob and extract the worker types in the job. Keeps the logic entirely in the job